src/lavinmq/http/handler/basic_auth.cr

```diff
@@ -23,24 +23,45 @@ module LavinMQ
           if auth.starts_with? "Basic "
             base64 = auth[6..]
             begin
-              decode = Base64.decode_string(base64)
-              if idx = decode.index(':')
-                username = decode[0...idx]
-                password = decode[idx + 1..]
-                if user = @server.users[username]?
-                  if valid_auth?(user, password) && guest_only_loopback?(context, user)
-                    context.authenticated_username = username
-                    return call_next(context)
-                  end
-                end
+              username = validate_auth(base64, context)
+              if username
+                context.authenticated_username = username
+                return call_next(context)
               end
             rescue Base64::Error
             end
           end
         end
+
+        if context.request.cookies["m"]?
+          base64 = context.request.cookies["m"].value.split("|")[1].split(":")[1]
```

Comment:
all this splitting is unnecessarily expensive. and is it always correct? always the second element?

Reply:
Yes, we can make it quicker!
And yes, since we set the cookie ourselves, so we know it always looks the same. 

I did some tests and `base64 = m.value.delete_at(0, (m.value.rindex(":") || 0) + 1)` should be ~2.5x quicker and use a lot less memory.

Reply:
nice! now you're getting into the mindset :) 

or a little bit more readable imho:

```
if idx = m.value.index(':')
  base64 = m.value[idx + 1..]
else
  raise "invalid cookie"
end
```